x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_logs/filter_dataset.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { memo, useState, useEffect } from 'react';
+import { EuiPopover, EuiFilterButton, EuiFilterSelectItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useStartServices } from '../../../../../hooks';
+import { AGENT_LOG_INDEX_PATTERN, DATASET_FIELD, AGENT_DATASET } from './constants';
+
+export const DatasetFilter: React.FunctionComponent<{
+  selectedDatasets: string[];
+  onToggleDataset: (dataset: string) => void;
+}> = memo(({ selectedDatasets, onToggleDataset }) => {
+  const { data } = useStartServices();
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [datasetValues, setDatasetValues] = useState<string[]>([AGENT_DATASET]);
+
+  useEffect(() => {
+    const fetchValues = async () => {
+      setIsLoading(true);
+      try {
+        const values = await data.autocomplete.getValueSuggestions({
+          indexPattern: {
+            title: AGENT_LOG_INDEX_PATTERN,
+            fields: [DATASET_FIELD],
+          },
+          field: DATASET_FIELD,
+          query: '',
+        });
+        setDatasetValues(values.sort());
+      } catch (e) {
+        setDatasetValues([AGENT_DATASET]);
+      }
+      setIsLoading(false);
+    };
+    fetchValues();
+  }, [data.autocomplete]);
+
+  return (
+    <EuiPopover
+      button={
+        <EuiFilterButton
```

Comment:
I noticed in the GIF that the filter button's popover includes padding. Add
`panelPaddingSize="none"` to remove it

Reply:
And we should do that for all filter button popovers

Reply:
fixed, thanks!